### PR TITLE
Sync with latest changes subql node

### PR DESCRIPTION
**Pull request by jiqiang90**
# Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have tested locally
- [x] I have performed a self review of my changes
- [ ] Updated any relevant documentation
- [ ] Linked to any relevant issues
- [ ] I have added tests relevant to my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] My code is up to date with the base branch

### CODE REVIEW

File: packages/node/src/configure/SubqueryProject.ts [hunk 1]
```diff
@@ -152,4 +165,58 @@ async function loadProjectTemplates(
...
+): Promise<SubqlProjectDs[]> {
+  const cron = new Cron();
+
+  dataSources = await Promise.all(
+    dataSources.map(async (ds) => {
+      if (isRuntimeDs(ds)) {
+        const startBlock = ds.startBlock ?? 1;
+        let block: AlgorandBlock;
+        let timestampReference: Date;
+        const safeApi = api.getSafeApi(startBlock);
```
Comment (jiqiang90) [inline packages/node/src/configure/SubqueryProject.ts]: @stwiname please review this, not sure we should use safeApi here

Comment (stwiname) [reply packages/node/src/configure/SubqueryProject.ts]: We can just use the normal api here.

File: packages/node/src/indexer/indexer.manager.ts [hunk 2]
```diff
@@ -164,15 +166,19 @@ export class IndexerManager {
...
   async start(): Promise<void> {
     await this.projectService.init();
+    logger.info('indexer manager started');
   }
 
   private filterDataSources(nextProcessingHeight: number): SubqlProjectDs[] {
-    let filteredDs = this.project.dataSources.filter(
+    let filteredDs: SubqlProjectDs[];
+
+    filteredDs = this.projectService.dataSources.filter(
```
Comment (stwiname) [inline packages/node/src/indexer/indexer.manager.ts]: Can be reverted?

File: packages/node/src/indexer/worker/block-dispatcher.service.ts [hunk 3]
```diff
@@ -34,6 +35,7 @@ import {
 } from './worker';
 
 const NULL_MERKEL_ROOT = hexToU8a('0x00');
+const { argv } = yargsOptions;
```
Comment (stwiname) [inline packages/node/src/indexer/worker/block-dispatcher.service.ts]: Can go. We should be doing everything through NodeConfig now

File: packages/node/src/indexer/worker/worker.service.ts [hunk 4]
```diff
@@ -51,11 +52,13 @@ export class WorkerService {
           this.fetchedBlocks[height] = block;
         }
 
-        // Return info to get the runtime version, this lets the worker thread know
+        const block = this.fetchedBlocks[height];
```
Comment (stwiname) [inline packages/node/src/indexer/worker/worker.service.ts]: I don't think this is needed?

**Comment by stwiname** [comment]
Tests are failing

**Review by stwiname — APPROVED** [review]

**Pull request merged**